File: Sem3/Advanced Methods of Programming/MyInterpreter - Java Console Application/project/src/view/TextMenu.java
package view;


import java.util.Map;
import java.util.Scanner;
import java.util.TreeMap;


public class TextMenu
{
    private Map<String, Command> commands;


    public TextMenu()
    {
        commands = new TreeMap<>();
    }


    public void addCommand(Command command)
    {
        commands.put(command.getKey(), command);
    }


    private void printMenu()
    {
        System.out.println("\nMenu:");
        for (Command command : commands.values())
        {
            String line = String.format("%4s. %s", command.getKey(), command.getDescription());
            System.out.println(line);
        }
    }


    public void show()
    {
        Scanner scanner = new Scanner(System.in);
        while (true)
        {
            printMenu();
            System.out.print("Input command: ");
            String key = scanner.nextLine();
            Command command = commands.get(key);
            if (command == null)
            {
                System.out.println("Invalid command.");
                continue;
            }
            command.execute();
        }
    }
}

File: Sem3/Advanced Methods of Programming/MyInterpreter - CSharp Console Application/MyInterpreter CSharp/view/TextMenu.cs
using System;
using System.Collections.Generic;

namespace MyInterpreter_CSharp.view
{
    public class TextMenu
    {
        private Dictionary<string, Command> _commands;

        public TextMenu()
        {
            _commands = new Dictionary<string, Command>();
        }

        public void AddCommand(Command command)
        {
            _commands.Add(command.Key, command);
        }

        private void PrintMenu()
        {
            Console.WriteLine("\nMenu:");
            foreach (Command cmd in _commands.Values)
            {
                string line = cmd.Key + ". " + cmd.Description;
                Console.WriteLine(line);
            }
        }

        public void Show()
        {
            while (true)
            {
                PrintMenu();
                Console.WriteLine("Input command: ");
                string key = Console.ReadLine();
                if (key == null || !_commands.TryGetValue(key, out Command command))
                {
                    Console.WriteLine("Invalid command.");
                    continue;
                }

                command.Execute();
            }
        }
    }
}